apps/web/src/app/(dashboard)/documents/data-table-action-dropdown.tsx

```diff
@@ -160,13 +159,20 @@ export const DataTableActionDropdown = ({ row }: DataTableActionDropdownProps) =
         />
       </DropdownMenuContent>
 
-      {isDocumentDeletable && (
+      {isDocumentDeletable && isDraft ? (
         <DeleteDraftDocumentDialog
           id={row.id}
           open={isDeleteDialogOpen}
           onOpenChange={setDeleteDialogOpen}
         />
+      ) : (
+        <DeleteDocumentDialog
+          id={row.id}
+          open={isDeleteDialogOpen}
+          onOpenChange={setDeleteDialogOpen}
+        />
       )}
+
```

Comment:
Feels like we could have just one dialog that conditionally requires the extra step based on the document status.